.github/workflows/docker-build-and-push-main.yaml

```diff
@@ -1,17 +1,32 @@
+# EVENTS: push, schedule, workflow_dispatch
+# workflow_dispatch: Build all. No publish to registry; save as tarball.
+# schedule: Build only devel and prebuilt. TAGS: date, latest
+# push-branch: Build only devel and prebuilt. TAGS: date, latest
+# push-tag: Build all. TAGS: version, date, latest
+
 name: docker-build-and-push-main
 
 on:
   push:
     tags:
-      - v*
-      - "[0-9]+.[0-9]+*"
+      - adkit-v*.*.*
+    branches:
+      - main
```

Comment:
Before this PR, `docker-build-and-push` was being triggered only on schedule.

After this change, and still, this action is running and pushing [(There are almost daily images published.)](https://github.com/autowarefoundation/autoware/pkgs/container/autoware/versions) per PR.

Was this expected behavior? Do we want the images to be updated this often?

![image](https://github.com/autowarefoundation/autoware/assets/10751153/893f922b-1e5e-4c09-9ee9-849f439ca6c0)

@oguzkaganozt @youtalk

Reply:
Let's discuss in:
- https://github.com/autowarefoundation/autoware/issues/4878